gtsam/linear/GaussianFactorGraph.h

```diff
@@ -180,20 +180,84 @@ namespace gtsam {
     ///@name Linear Algebra
     ///@{
 
+    /// Sparse matrix representation as vector of tuples.
+    typedef std::vector<boost::tuple<size_t, size_t, double>>
+        SparseMatrixBoostTriplets;
+    /// Sparse matrix representation as vector of slightly more efficient
+    /// tuples.
+    typedef std::vector<std::tuple<int, int, double>> SparseMatrixGtsamTriplets;
```

Comment:
What's the reasoning here? I'd prefer to keep only one of these.

Reply:
Originally the code used `SparseMatrixBoostTriplets`.  I added `SparseMatrixGtsamTriplets` because it's about 14% faster and 33% less memory intensive (16 bytes instead of 24 bytes per nnz entry) in my tests.  Reason: 1) `size_t` is 8 bytes, and even if we use int instead, `boost::tuple` aligns to 8 bytes.  We don't have to typedef these?

Old version kept for backwards compatibility.

Reply:
I removed the typedefs but both are still used (the std::tuple version is used in the "fast" version).